### PR TITLE
Get all transactions

### LINKED ISSUE #15: getAllTransactions
- Endpoint URI: */transactions*
- Http Method: *GET*
- API Logic: 
    - get all transactions
   - map event name
- Progress
    - [x] development
    - [x] integration Test

### LINKED ISSUE #26: get All Transactions By EventId
- Endpoint URI: */transactions/event/{eventId}*
- Http Method: *GET*
- API Logic: 
    - check event is not cancelled
    - get all transactions of the event
- Progress
    - [x] development
    - [x] integration Test

### PR DESCRIPTION
**Pull request by HowardYaw**
- resolve #15
- resolve #26
- comment out unused API and respective test

### CODE REVIEW

File: src/main/java/com/thirdcc/webapp/service/dto/TransactionDTO.java [hunk 1]
```diff
@@ -30,6 +32,10 @@
 
     private Instant lastModifiedDate;
 
+    private String eventName;
+
+    private Receipt receiptDetails;
```
Comment (kachun333) [inline src/main/java/com/thirdcc/webapp/service/dto/TransactionDTO.java]: I have this idea, I'm thinking that we can standardize the naming convention for nested object in DTO
```
private Receipt receiptDetails;
```
---
```
private Receipt receiptDTO;
```
what do you think? @HowardYaw

**Review by kachun333 — DISMISSED** [review]

File: src/main/java/com/thirdcc/webapp/service/dto/TransactionDTO.java (hunk 1 above)
Comment (HowardYaw) [reply src/main/java/com/thirdcc/webapp/service/dto/TransactionDTO.java]: You means standardised to receiptDTO? {entity}DTO

Comment (kachun333) [reply src/main/java/com/thirdcc/webapp/service/dto/TransactionDTO.java]: yes. what do you think?

Comment (HowardYaw) [reply src/main/java/com/thirdcc/webapp/service/dto/TransactionDTO.java]: It is better
i will make the changes

Comment (HowardYaw) [reply src/main/java/com/thirdcc/webapp/service/dto/TransactionDTO.java]: d00fc12d33d9ecf934ccaa3cbea243b8c7771880
@kachun333 please verify

**Review by kachun333 — DISMISSED** [review]

**Review by CarlosSia — DISMISSED** [review]

File: src/main/java/com/thirdcc/webapp/service/impl/TransactionServiceImpl.java [hunk 2]
```diff
@@ -86,4 +102,19 @@ public void delete(Long id) {
         log.debug("Request to delete Transaction : {}", id);
         transactionRepository.deleteById(id);
     }
+
+    @Override
+    public Page<TransactionDTO> findAllByEventId(Long eventId, Pageable pageable) {
+        log.debug("Request to find all Transaction of event: {]", eventId);
```
Comment (CarlosSia) [inline src/main/java/com/thirdcc/webapp/service/impl/TransactionServiceImpl.java]: got typo in log debug, the close bracket is } not ]

And for this pr is not checking whether the user is event crew or admin or not right?

Comment (HowardYaw) [reply src/main/java/com/thirdcc/webapp/service/impl/TransactionServiceImpl.java]: resolve in 6b9182207f9785d5608b022dfdf8e229b708f74d

**Review by kachun333 — APPROVED** [review]

**Review by CarlosSia — APPROVED** [review]

**Pull request merged**